appserver/tests/tck/microprofile/rest-client/tck-suite.xml

```diff
@@ -0,0 +1,35 @@
+<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >
+<!--
+
+    Copyright (c) 2023 Contributors to the Eclipse Foundation. All rights reserved.
+
+    This program and the accompanying materials are made available under the
+    terms of the Eclipse Public License v. 2.0, which is available at
+    http://www.eclipse.org/legal/epl-2.0.
+
+    This Source Code may also be made available under the following Secondary
+    Licenses when the conditions for such availability set forth in the
+    Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+    version 2 with the GNU Classpath Exception, which is available at
+    https://www.gnu.org/software/classpath/license.html.
+
+    SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+
+-->
+
+<suite name="MicroProfile Rest Client TCK" verbose="2" configfailurepolicy="continue" >
+
+    <test name="MicroProfile Rest Client TCK">
+        <packages>
+            <package name="org.eclipse.microprofile.rest.client.tck.*" />
+        </packages>
+        <classes>
+            <class name="org.eclipse.microprofile.rest.client.tck.cditests.ConfigKeyForMultipleInterfacesTest">
+                <methods>
+                    <exclude name="testConfigKeyUsedForUri" />
```

Comment:
I don't know yet why this test fails ;(

Reply:
@arjantijms @hs536 @OndroMih Any ideas?

Reply:
So, only in this test provider setup done with fqcn_od_client/mp-rest/provider=fqcn_of_provider property in the `microprofile-config.properties` file. Config file resides in `WEB-INF/lib/test_name.jar/META-INF` directory of the test war and contains also another property. This other property reads as expected. It seems to like a provider property reads as a "regular" property, not as provider, in this paticular case, and thus provider not registered.

Nevertheless, standalone sample application with same structure, deployed to GlassFish Server, works as expected.

It seems to like that this problem is only related to the test environment.

If we place the `microprofile-config.properties` file with provider property to the `WEB-INF\classes\META-INF` directory, all works as expected.

Reply:
**UPDATE:** Test fails because both `ConfigKeyTest` and `ConfigKeyForMultipleInterfacesTest` deploys war with the same name `ConfigKeyTest.war`

Reply:
I raised this https://github.com/eclipse/microprofile-rest-client/issues/363. You could raise a PR against the TCK.

As a workaround, maybe we could change the archive name in an Arquillian transformer to always add some random numbers to the deployment file?

Reply:
Thank you for reply, @OndroMih.

Latest GlassFish Arquillian managed server plugin contains needed ability.